core/domain/email_manager_test.py

```diff
@@ -1326,3 +1326,108 @@ def test_that_feedback_message_emails_are_correct(self):
+
+        self.set_moderators([self.moderator2_username, self.MODERATOR_USERNAME])
+
+        self.exploration = self.save_new_default_exploration(
+            'A', self.editor_id, 'Title')
+        self.owner_ids = [self.editor_id]
+
+        self.report_text = 'AD'
+
+        self.can_send_emails_ctx = self.swap(
+            feconf, 'CAN_SEND_EMAILS', True)
+
+    def test_that_flag_exploration_emails_are_correct(self):
+
+        expected_email_subject = 'Exploration flagged by user: "Title"'
+
+        expected_email_html_body = (
+            'Hello Moderator,<br>'
+            'newuser has submitted a new report on the exploration Title'
+            ' on the grounds of: <br>'
+            'AD .<br>'
+            'You can modify the exploration by clicking '
+            '<a href="https://www.oppia.org/create/A">'
+            '"here"</a>.<br>'
+            '<br>'
+            'Thanks!<br>'
+            '- The Oppia Team<br>'
+            '<br>'
+            'You can change your email preferences via the '
+            '<a href="https://www.example.com">Preferences</a> page.')
+
+        expected_email_text_body = (
+            'Hello Moderator,\n'
+            'newuser has submitted a new report on the exploration Title'
+            ' on the grounds of: \n'
+            'AD .\n'
+            'You can modify the exploration by clicking "here".\n'
+            '\n'
+            'Thanks!\n'
+            '- The Oppia Team\n'
+            '\n'
+            'You can change your email preferences via the Preferences page.')
+
+        with self.can_send_emails_ctx:
+            email_manager.send_flag_exploration_email(
+                self.exploration.title, self.exploration.id, self.new_user_id,
+                self.report_text)
+
+            # Make sure correct email is sent.
+            messages = self.mail_stub.get_sent_messages(to=self.MODERATOR_EMAIL)
+            self.assertEqual(len(messages), 1)
+            self.assertEqual(
+                messages[0].html.decode(),
+                expected_email_html_body)
+            self.assertEqual(
+                messages[0].body.decode(),
+                expected_email_text_body)
+
+            # Make sure correct email is sent to multiple moderators.
+            messages = self.mail_stub.get_sent_messages(
+                to=self.moderator2_email)
+            self.assertEqual(len(messages), 1)
+            self.assertEqual(
+                messages[0].html.decode(),
+                expected_email_html_body)
+            self.assertEqual(
+                messages[0].body.decode(),
+                expected_email_text_body)
+
+            # Make sure correct email models are stored.
+            all_models = email_models.SentEmailModel.get_all().fetch()
+            sent_email_model = all_models[0]
+            self.assertEqual(
+                sent_email_model.subject, expected_email_subject)
+            #self.assertEqual(
```

Comment:
Hey @seanlip , There seems to be a little issue here. This test runs successfully sometimes and fails sometimes, when the following four lines are uncommented.
The value of sent_email_model.recipient_id is moderator_id, sometimes and moderator2_id, other times. I think this is due to the order of tasks handled by taskqueue. What do you think/ suggest?